TF-TRT Enhance README

pooyadavoodi opened the pull request:
This enhances the readme file in TF-TRT with some links to examples and docs.

Commit:
Update README.md

Commit:
Update README.md

aaroey reviewed (dismissed):
Thanks for enhancing this!

rversaw commented on tensorflow/contrib/tensorrt/README.md:
```diff
@@ -1,8 +1,51 @@
+of TensorRT from the
+[NVIDIA Developer website](https://developer.nvidia.com/tensorrt).
+Installation instructions for compatibility with TensorFlow are provided on the
+[TensorFlow GPU support](https://www.tensorflow.org/install/gpu) guide.
```
This portion is pretty similar to what it used to be, but it does raise a few questions, so clarifications could be useful. It used to reference `TensorRT 3.0.4`, which could be beneficial if an exact version is necessary. At the same time, the `NVIDIA Developer website` points you to `TensorRT 5` (is that supported yet?), and the `TensorFlow GPU support` page has a command to install `libnvinfer4=4.1.2-1+cuda9.0`.

I don't know where the definitive version should be documented, but right now it's a little unclear what is necessary as a user - If I try to follow these instructions directly, I'll end up with TensorRT 5 in addition to `libnvinfer4`, which doesn't seem like an ideal setup. If the `sudo apt install libnvinfer4=4.1.2-1+cuda9.0` gives me all the dependencies I need for this, I think linking to the `TensorFlow GPU support` as the primary resource would be the most intuitive, while the NVIDIA site could be supplementary.

Again, this PR doesn't change these instructions much, so this could be clarified outside of this PR, but as a user who has ran into confusion on this, some clarifications in the docs would be appreciated!

pooyadavoodi replied:
I removed the NVIDIA link encouraging users to start from https://www.tensorflow.org/install/gpu for installing TensorRT to reduce the confusion about which TensorRT version to install. I hope that makes it a bit better. Thanks for pointing this out.

rversaw commented on tensorflow/contrib/tensorrt/README.md:
```diff
@@ -1,8 +1,51 @@
+
+## Installing TF-TRT
+
+Currently Tensorflow nightly builds include TF-TRT by default,
```
`Tensorflow`->`TensorFlow`?

pooyadavoodi replied:
Thanks @rversaw. I fixed it.

rversaw commented on tensorflow/contrib/tensorrt/README.md:
```diff
@@ -1,8 +1,51 @@
+install the latest TF pip package to get access to the latest TF-TRT.
+
+If you want to use TF-TRT on NVIDIA Jetson platform, you can find
+the download links for the relevant Tensorflow pip packages here:
```
Another `Tensorflow`->`TensorFlow`?

pooyadavoodi replied:
Thanks @rversaw. I fixed it.

rversaw commented on tensorflow/contrib/tensorrt/README.md:
```diff
@@ -1,8 +1,51 @@
+
+TF-TRT includes both Python tests and C++ unit tests.
+Most of Python tests are located in the test directory
+and they can be executed uring `bazel test` or directly
```
`uring`->`using`

pooyadavoodi replied:
Thanks @rversaw. I fixed it.

Commit:
Update README.md

Tensorflow -> TensorFlow

uring -> using

Commit:
Update README.md

Remove NVIDIA link encouraging users to start from https://www.tensorflow.org/install/gpu for installing TensorRT to reduce the confusion about which TensorRT version to install.

smit-hinsu commented on tensorflow/contrib/tensorrt/README.md:
```diff
@@ -1,8 +1,50 @@
+
+## Documentation
+
+[TF-TRT documentaion](https://docs.nvidia.com/deeplearning/dgx/integrate-tf-trt/index.html)
```
s/documentaion/documentation

pooyadavoodi replied:
Thanks @smit-hinsu. Fixed it.

I just copied the doc to a Google doc to make sure I don't have any more typos!

smit-hinsu reviewed (dismissed):
Thanks for upgrading the documentation.

I wonder if we should keep user documentation in a single place with README file linking to that documentation. That way we don't need to worry about keeping both versions in sync and this file need to only address developers planning to contribute to TF-TRT. Not required in this PR but might make sense if this documentation becomes larger.

Commit:
Update README.md

documentaion --> documentation

pooyadavoodi commented:
@smit-hinsu I was also thinking of not extending this doc anymore. Once contrib goes away, it will be automatically removed.

We also have https://github.com/tensorflow/tensorrt/blob/master/README.md which is currently more or less a duplicate of this one. That will be improved and extended over time.
